crates/neon/src/sys/object.rs

```diff
@@ -113,20 +112,14 @@ pub unsafe fn set_string(
 
     *out = true;
 
-    match napi::create_string_utf8(env, key as *const _, len as usize, key_val.as_mut_ptr()) {
-        Err(_) => {
-            *out = false;
-            return false;
-        }
-        Ok(()) => (),
+    if napi::create_string_utf8(env, key as *const _, len as usize, key_val.as_mut_ptr()).is_err() {
+        *out = false;
+        return false;
     }
 
-    match napi::set_property(env, object, key_val.assume_init(), val) {
-        Err(_) => {
-            *out = false;
-            return false;
-        }
-        Ok(()) => (),
+    if napi::set_property(env, object, key_val.assume_init(), val).is_err() {
```

Comment:
I think a lot of these might be subtly incorrect. The legacy backend assumed exception in many places when `false` was returned, but that might not be the case.

The correct thing to do might be to return `false` on exception and otherwise panic.